src/lib/sw_engine/tvgSwRenderer.cpp

```diff
@@ -131,7 +129,10 @@ struct SwShapeTask : SwTask
                 if (auto fill = sdata->strokeFill()) {
                     auto ctable = (flags & RenderUpdateFlag::GradientStroke) ? true : false;
                     if (ctable) shapeResetStrokeFill(&shape);
-                    if (!shapeGenStrokeFillColors(&shape, fill, transform, surface, opacity, ctable)) goto err;
```

Comment:
In this context, stroking is always true

Contextual function calls generates more cpu instructions, you can write it one line.
shapeGenStrokeFillColors(........, (filling ? 255 : opacity), ...) goto err;
or
auto opacity2 = filling ? 255 : opacity;
shapeGenStrokeFillColors(........, opacity2, ...) goto err;